### ISSUE TITLE
Look into black screen upon leaving score screen after exiting map

### ISSUE THREAD

**Issue by nvs** [nvs/gem#480]
It certainly is a bug. It did not occur in 1.30. And it happens in both the 1.5 and 1.6 series of the maps. The black screen is not significant in 1.4. As far as 1.31 being the culprit, it most certainly is. The question is whether I can resolve the issue within the map.

## Testing
*This list will be updated as I test more things.*

- Memory leaks in general. A large memory leak seems to replicate the experienced behavior. However, memory usage in Gem does not increase in a fashion that would suggest one.
- The string table in Jass. Every string that is unique is added to this table. In 1.30, a table of 450,000 entries would not cause a significant delay upon exiting the score screen. In 1.31, a table of such size causes a significant delay. This certainly could present an issue for Gem. Although, I would need to identify how many strings Gem is utilizing. For reference, disabling the board in Gem did not resolve the issue in offline play.
- Units could be a potential issue if they are active (i.e. alive or in the process of decaying). That is, they are not removed from the map. Removed units present no issues. Testing shows that having ~12,000 active units when exiting will cause a significant delay upon leaving the score screen (but not nearly enough to equal some of the durations experienced). However, this should never really be an issue in Gem as there are probably never more than 500 active units at any given time.
- Hashtables do not seem present an issue. Approaching 600,000 parent keys in a single hashtable will crash the client due to lack of memory. Each parent key is its own unique list. Having 3,000,000 child keys within a single parent key presents no issue. Neither case causes any delay upon leaving the score screen.
- The sheer act of starting timers seems to contribute as well. This behavior is exhibited in Gem's clock that uses a timer dialog. A timer is started and immediately paused multiple times a second. However, this does not seem to happen enough to reach black screen durations achieved in testing. A periodic timer does not contribute, as it seems to only 'start' once.

**Comment by nvs** [comment]
TL;DR: Not worth the effort to diagnostic what could be Jass specific issues. And even then, doing the work in Jass instead of just rewriting in Lua is ultimately time wasted. I'll leave this open in case a smoking gun appears. But no more effort will be spent researching this.

**Comment by nvs** [comment]
Okay, I lied. I'll look into it a bit more since it did not occur (much) in 1.4.

**Comment by nvs** [comment]
This is no longer an issue as of 1.32.0.14481.
